src/main/java/school/lemon/changerequest/java/extendedinteger/ExtendedInteger.java

```diff
@@ -4,116 +4,178 @@
  * Integer representation with some additional features
  */
 public class ExtendedInteger {
+    private int value;
 
     public ExtendedInteger(int value) {
-        //TODO: implement me
+        this.value = value;
     }
 
     /**
      * Check whether specified value is even
+     *
      * @param value to check
      * @return true if value is even, false - otherwise
      */
     public static boolean isEven(int value) {
-        //TODO: implement me
+        if ((value & 1) == 0) {
+            return true;
+        }
         return false;
     }
 
     /**
      * Check whether specified value is odd
+     *
      * @param value to check
      * @return true if value is odd, false - otherwise
      */
     public static boolean isOdd(int value) {
-        //TODO: implement me
+        if ((value & 1) == 1) {
+            return true;
+        }
         return false;
     }
 
     /**
      * Check whether specified value is prime
+     *
      * @param value to check
      * @return true if value is prime, false - otherwise
      */
     public static boolean isPrime(int value) {
-        //TODO: implement me
-        return false;
+        if (value <= 1) {
+            return false;
+        } else {
+            for (int i = 2; i < value; i++) {
+                if (value % i == 0) {
+                    return false;
+                }
+            }
+        }
+        return true;
     }
 
     /**
      * Parse specified char array and create instance of {@code ExtendedInteger}
+     *
      * @param value to parse
      * @return instance of {@code ExtendedInteger} or
      * null in case specified value is null or the value does not contain a parsable integer
      */
     public static ExtendedInteger parseInt(char[] value) {
-        //TODO: implement me
-        return null;
+
+        int result = 0;
+        int resultSum = 0;
+        int j = 0;
+        int k = 0;
+        int first = 1;
+        int lengthChar = value.length;
+        if (lengthChar == 0) {
+            return null;
+        }
+        if (value[0] == '-') {
+            first = -1;
+        } else if (value[0] == '+') {
+            k = 0;
+        } else {
+            k = -1;
+        }
+        for (int i = lengthChar - 1; i != k; i--) {
+            if (Character.isDigit(value[i])) {
+                result = Character.getNumericValue(value[i]);
+                result *= Math.pow(10, j++);
+            } else {
+                return null;
+            }
+            resultSum += result;
+        }
+        resultSum *= first;
```

Comment:
You algorithm is OK, but implementation details sucks.
Try to create more valuable variable names, for example:
What do you mean with your `first` variable ? In fact, you need to check the sign of int value, if it's plus or minus and multiply result by 1 or -1 respectively. So, probably variable name `sign` will suite here better.
Additionally you've got `j`, `k`, `i` indexes, but you really need only one.
Also, there is no need for 2 variables for `result` you can reuse one.

But, anyway, you code do pass all the tests.